examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c

```diff
@@ -655,14 +721,169 @@ static sl_status_t wfx_rsi_do_join(void)
+    }
+}
+
+void ProcessEvent(WfxEvent_t inEvent)
+{
+    WfxEvent_t outEvent;
+    // Process event
+    switch (inEvent.eventType)
+    {
+    case WFX_EVT_STA_CONN:
+        SILABS_LOG("%s: starting LwIP STA", __func__);
+        wfx_rsi.dev_state |= WFX_RSI_ST_STA_CONNECTED;
+        outEvent.eventType = WFX_EVT_STA_DO_DHCP;
+        WfxPostEvent(&outEvent);
+        hasNotifiedWifiConnectivity = false;
+#if (CHIP_DEVICE_CONFIG_ENABLE_IPV4)
+        hasNotifiedIPV4 = false;
+#endif // CHIP_DEVICE_CONFIG_ENABLE_IPV4
+        hasNotifiedIPV6 = false;
+        wfx_lwip_set_sta_link_up();
+        /* We need to get AP Mac - TODO */
+        // Uncomment once the hook into MATTER is moved to IP connectivty instead
+        // of AP connectivity.
+        // wfx_connected_notify(0, &wfx_rsi.ap_mac); // This
+        // is independant of IP connectivity.
+        break;
+    case WFX_EVT_STA_DISCONN:
+        // TODO: This event is not being posted anywhere, seems to be a dead code or we are missing something
+        wfx_rsi.dev_state &=
+            ~(WFX_RSI_ST_STA_READY | WFX_RSI_ST_STA_CONNECTING | WFX_RSI_ST_STA_CONNECTED | WFX_RSI_ST_STA_DHCP_DONE);
+        SILABS_LOG("%s: disconnect notify", __func__);
+        /* TODO: Implement disconnect notify */
+        wfx_lwip_set_sta_link_down(); // Internally dhcpclient_poll(netif) ->
+                                      // wfx_ip_changed_notify(0) for IPV4
+#if (CHIP_DEVICE_CONFIG_ENABLE_IPV4)
+        wfx_ip_changed_notify(IP_STATUS_FAIL);
+        hasNotifiedIPV4 = false;
+#endif /* CHIP_DEVICE_CONFIG_ENABLE_IPV4 */
+        wfx_ipv6_notify(GET_IPV6_FAIL);
+        hasNotifiedIPV6    = false;
+        outEvent.eventType = WFX_EVT_STA_DO_DHCP;
+        WfxPostEvent(&outEvent);
+        hasNotifiedWifiConnectivity = false;
+        break;
+    case WFX_EVT_AP_START:
+        // TODO: Currently unimplemented
+        break;
+    case WFX_EVT_AP_STOP:
+        // TODO: Currently unimplemented
+        break;
+    case WFX_EVT_SCAN:
+#ifdef SL_WFX_CONFIG_SCAN
+        if (!(wfx_rsi.dev_state & WFX_RSI_ST_SCANSTARTED))
+        {
+            SILABS_LOG("%s: start SSID scan", __func__);
+            sl_wifi_scan_configuration_t wifi_scan_configuration = { 0 };
+
+            // TODO: Add scan logic
+            sl_wifi_advanced_scan_configuration_t advanced_scan_configuration = { 0 };
+            int32_t status;
+            advanced_scan_configuration.active_channel_time  = ADV_ACTIVE_SCAN_DURATION;
+            advanced_scan_configuration.passive_channel_time = ADV_PASSIVE_SCAN_DURATION;
+            advanced_scan_configuration.trigger_level        = ADV_SCAN_THRESHOLD;
+            advanced_scan_configuration.trigger_level_change = ADV_RSSI_TOLERANCE_THRESHOLD;
+            advanced_scan_configuration.enable_multi_probe   = ADV_MULTIPROBE;
+            status = sl_wifi_set_advanced_scan_configuration(&advanced_scan_configuration);
```

Comment:
The `status` is assigned here but not checked, is that OK?

Reply:
Check added, followup for actions to take on failure.